docs/algorithms/on_the_fly_circuit_design_on_defective_surface.rst

```diff
@@ -0,0 +1,39 @@
+.. _on_the_fly_design:
+
+SiDB Circuit Design Algorithm in the presence of Atomic Defects
+---------------------------------------------------------------
+
+**Header:** ``fiction/algorithms/physical_design/on_the_fly_circuit_design_on_defective_surface.hpp``
+
+This algorithm is designed to create SiDB circuits on a clocked surface, accommodating the presence of atomic defects.
+
+1. **Blacklist Generation**:
+
+   Initially, a blacklist of gate-tile pairs is generated. This blacklist is based on the locations of neutrally charged atomic defects and their overlap with the I/O pins of the SiDB skeletons.
+
+2. **Gate-Level Layout Design**:
+
+   Using the generated blacklist, a gate-level layout is designed with the ``exact`` algorithm. This process involves:
+
+   - **Valid Layout Found**:
+
+     If a valid gate-level layout is found, the corresponding gates are designed.
+
+   - **Invalid Layout**:
+
+     If a valid layout is not found, the blacklist is updated, and the placement and routing process is repeated.
+
+This iterative approach ensures that the designed SiDB circuits can effectively handle defects present on the surface.
+
+
+.. tabs::
+    .. tab:: C++
+        **Header:** ``fiction/algorithms/physical_design/on_the_fly_circuit_design_on_defective_surface.hpp``
+
+        .. doxygenstruct:: fiction::on_the_fly_circuit_design_params
+           :members:
+        .. doxygenstruct:: fiction::on_the_fly_circuit_design_stats
+            :members:
+        .. doxygenfunction:: fiction::on_the_fly_circuit_design_on_defective_surface(const Ntk& ntk, const GateLyt& lattice_tiling,
```

Comment:
@marcelwa It fails in the doc generator. I have also noticed that even the docu of ``exact`` shows problems on ``main``.

Reply:
This works now, right?

Reply:
Yes, it does, but there are still problems with ``exact``

Reply:
Yes, I haven't updated the `exact` docs in this PR because it is unrelated. Unless we made changes to `exact` here, too.

Reply:
No, I don't think so. It does not work on ``main`` either.